### PR TITLE
zig 0.10.0

### PR DESCRIPTION
**Pull request by chenrui333**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by chenrui333** [comment]
`cargo-zigbuild` test failure, I could not reproduce in my local.

**Comment by messense** [comment]
> `cargo-zigbuild` test failure, I could not reproduce in my local.

Looks like the error is coming from `semver`: https://github.com/dtolnay/semver/blob/3a2fd6f8aa7c6d7dbc7550851bfd9c28767820a1/tests/test_version.rs#L17

What's the output of `zig version`?

File: Formula/zig.rb [hunk 1]
```diff
@@ -19,12 +15,9 @@ class Zig < Formula
...
   end
 
-  head do
-    url "https://github.com/ziglang/zig.git", branch: "master"
-    depends_on "llvm" => :build
-  end
-
   depends_on "cmake" => :build
+  depends_on "llvm" => :build
+
```
Comment (carlocab) [inline Formula/zig.rb:20]:
```suggestion
  depends_on "llvm" => :build
  depends_on "z3"
  depends_on "zstd"
```

**Comment by carlocab** [comment]
> `cargo-zigbuild` test failure, I could not reproduce in my local.

Fixed now.

However, it looks like we need `zig` to use `-headerpad_max_install_names` when linking. Not sure if there's a way to do this (or if it's even supported in `zld`).

**Comment by carlocab** [comment]
Ok, `zig` supports it as of 0.10: https://github.com/ziglang/zig/commit/589bf67635a9fe9c3e6df4b63c09e0cc4954d29a

Still need to work out how to get the build to use it though.

**Comment by carlocab** [comment]
Fails to build on Catalina. Support dropped. https://github.com/ziglang/zig/issues/13313

**Review by carlocab — DISMISSED** [review]

File: Formula/zig.rb [hunk 2]
```diff
@@ -19,12 +24,10 @@ class Zig < Formula
...
   end
 
-  head do
-    url "https://github.com/ziglang/zig.git", branch: "master"
-    depends_on "llvm" => :build
-  end
-
   depends_on "cmake" => :build
+  depends_on "llvm" => :build
+  depends_on "z3"
```
Comment (carlocab) [inline Formula/zig.rb:29]:
```suggestion
  depends_on "llvm" => :build
  depends_on macos: :big_sur # https://github.com/ziglang/zig/issues/13313
  depends_on "z3"
```

**Review by carlocab — DISMISSED** [review]

**Comment by lukeholder** [comment]
So is this ready to merge? 🙏🏻

**Comment by carlocab** [comment]
It will be when the bottles are built. Merging it right now means no bottles, and consequently `brew upgrade` and `brew install` errors for users.

**Review by carlocab — APPROVED** [review]

**Comment by smbl64** [comment]
```
brew reinstall z3
```
fixed the problem with `libz3.4.11.dylib` for me.

**Pull request closed without merging**